manifests/enterprise/dashboard/package.pp

```diff
@@ -0,0 +1,33 @@
+class sensu::enterprise::dashboard::package {
+
+  if $caller_module_name != $module_name {
+    fail("Use of private class ${name} by ${caller_module_name}")
+  }
+
+  if $::sensu::install_repo and $::sensu::enterprise_dashboard {
+    $ensure = 'present'
+
+    $se_user = $sensu::enterprise_user
+    $se_pass = $sensu::enterprise_pass
+
+    $url = "http://${se_user}:${se_pass}@enterprise.sensuapp.com/yum/\$basearch/"
+
+    yumrepo { 'sensu-enterprise-dashboard':
```

Comment:
I think the motivation was initially because of the potential complexity of the $repo_require logic (a few lines down), but I'll see if I can merge all the repo stuff in to a single class (per platform).